query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/ExecutionContextTest.java

```diff
@@ -308,4 +311,133 @@ public void testSelectionsLinkedHashSet() {
     assertEquals(
         Expression.newBuilder().setFunction(minFunction).build(), selectionsIterator.next());
   }
+
+  @Test
+  public void testSetTimeSeriesPeriod() {
+
+    // no group by
+    Builder builder = QueryRequest.newBuilder();
+    QueryRequest queryRequest = builder.build();
+    ExecutionContext context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+
+    // group by on column
+    builder.addGroupBy(
+        Expression.newBuilder()
+            .setColumnIdentifier(
+                ColumnIdentifier.newBuilder().setColumnName("Trace.service_name")));
+    queryRequest = builder.build();
+    context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+
+    // group by on other functions
+    Function count =
+        Function.newBuilder()
+            .setFunctionName("Count")
+            .setAlias("myCountAlias")
+            .addArguments(
+                Expression.newBuilder()
+                    .setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName("Trace.id")))
+            .build();
+    builder.addGroupBy(Expression.newBuilder().setFunction(count));
+    queryRequest = builder.build();
+    context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+  }
+
+  @Test
+  public void testSetTimeSeriesPeriodForTimeSeriesRequest() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addGroupBy(
+        Expression.newBuilder()
+            .setFunction(createTimeColumnGroupByFunction("SERVICE.startTime", 15))
+            .build());
+    QueryRequest queryRequest = builder.build();
+    ExecutionContext context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.of(Duration.ofSeconds(15)), context.getTimeSeriesPeriod());
+  }
+
+  @Test
+  public void testComputeTimeRangeDuration() {
+    QueryRequest queryRequest = getQueryRequestWithTimeFilter();
+    ExecutionContext context = new ExecutionContext("test", queryRequest);
+    context.setTimeFilterColumn("SERVICE.startTime");
+    context.computeTimeRangeDuration(queryRequest);
+    assertEquals(Duration.ofSeconds(3600), context.getTimeRangeDuration());
+  }
+
+  private static QueryRequest getQueryRequestWithTimeFilter() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addGroupBy(
+        Expression.newBuilder()
+            .setFunction(createTimeColumnGroupByFunction("SERVICE.startTime", 15))
+            .build());
+    Filter filter1 =
+        createFilter(
+            "SERVICE.startTime",
+            Operator.GE,
+            ValueType.LONG,
+            System.currentTimeMillis() - Duration.ofHours(1).toMillis());
```

Comment:
nit: no reason to use a dynamic value like `System.currentTimeMillis()` in this test. If we're not careful, these can often lead to cases of nondeterministic test results (such as the clock ticking between this line and 382). Could just as easily be a hardcoded value. e..g.
```java
    Instant.ofEpochSecond(SECONDS.convert(1, HOURS));
    Instant.ofEpochSecond(SECONDS.convert(2, HOURS));
```

Additionally, I think we have a util to create time range filter pair somewhere.